ast/src/lib.rs

```diff
@@ -3,9 +3,20 @@
 use itertools::Itertools;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
 use std::fmt::{Display, Result, Write};
 use std::sync::Arc;
 
+// TODO: Find a better place for this
+pub trait WitgenCallback<T>:
+    Fn(&[(String, Vec<T>)], BTreeMap<u64, T>, u8) -> Vec<(String, Vec<T>)>
```

Comment:
Maybe we could change this to a proper function?

Reply:
I made a struct that has a proper function!